contracts/AssetPool.sol

```diff
@@ -296,6 +325,113 @@ contract AssetPool is Ownable, IAssetPool {
+    ///         complete the withdrawal. The change needs to be finalized with
+    ///         a call to finalizeWithdrawalTimeoutUpdate after the required
+    ///         governance delay passes.
+    /// @param  _newWithdrawalTimeout The new value of the withdrawal timeout.
+    function beginWithdrawalTimeoutUpdate(uint256 _newWithdrawalTimeout)
+        external
+        onlyOwner
+    {
+        newWithdrawalTimeout = _newWithdrawalTimeout;
+        withdrawalTimeoutChangeInitiated = block.timestamp;
+        emit WithdrawalTimeoutUpdateStarted(
+            _newWithdrawalTimeout,
+            block.timestamp
+        );
+    }
+
+    /// @notice Lets the contract owner to finalize an update of withdrawal
+    ///         timeout parameter value. This call has to be preceded with
+    ///         a call to beginWithdrawalTimeoutUpdate and the governance delay
+    ///         has to pass.
+    function finalizeWithdrawalTimeoutUpdate()
+        external
+        onlyOwner
+        onlyAfterWithdrawalGovernanceDelay(withdrawalTimeoutChangeInitiated)
+    {
+        withdrawalTimeout = newWithdrawalTimeout;
+        emit WithdrawalTimeoutUpdated(withdrawalTimeout);
+        newWithdrawalTimeout = 0;
+        withdrawalTimeoutChangeInitiated = 0;
+    }
+
+    /// @notice Returns the remaining time that has to pass before the contract
+    ///         owner will be able to finalize withdrawal delay update.
+    ///         Bear in mind the contract owner may decide to wait longer and
+    ///         this value is just an absolute minimum.
+    function getRemainingWithdrawalDelayUpdateTime()
+        external
+        view
+        returns (uint256)
+    {
+        return
+            GovernanceUtils.getRemainingChangeTime(
+                withdrawalDelayChangeInitiated,
+                withdrawalGovernanceDelay()
+            );
+    }
+
+    /// @notice Returns the remaining time that has to pass before the contract
+    ///         owner will be able to finalize withdrawal timeout update.
+    ///         Bear in mind the contract owner may decide to wait longer and
+    ///         this value is just an absolute minimum.
+    function getRemainingWithdrawalTimeoutUpdateTime()
+        external
+        view
+        returns (uint256)
+    {
+        return
+            GovernanceUtils.getRemainingChangeTime(
+                withdrawalTimeoutChangeInitiated,
+                withdrawalGovernanceDelay()
+            );
+    }
+
+    /// @notice The time it takes to initiate and complete the withdrawal from
+    ///         the pool plus 2 days to make a decision. This governance delay
+    ///         should be used for all changes directly affecting underwriter
+    ///         positions. This time is a minimum and the governance may choose
+    ///         to wait longer before finalizing the update.
+    function withdrawalGovernanceDelay() public view returns (uint256) {
```

Comment:
Non-blocking (https://github.com/keep-network/coverage-pools/issues/82): Can we document the returned value using `@return`?

Reply:
https://github.com/keep-network/coverage-pools/pull/105/commits/c0764c44ac0bc0884c60130ce06bce5fefa46cb8